packages/server/src/modules/retail-location/retail-location.resolver.ts

```diff
@@ -133,4 +137,242 @@ export class RetailLocationResolver {
+      retailLocationId,
+      message: "You do not have permission to view these reservations.",
+    });
+
+    const getTotalBooksWithProblems = async () => ({
+      totalBooksWithProblems: await this.prisma.bookCopy.count({
+        where: {
+          book: {
+            retailLocationId,
+          },
+          problems: {
+            some: {
+              resolvedAt: null,
+            },
+          },
+          returnedAt: {
+            not: null,
+          },
+        },
+      }),
+    });
+
+    const getTotalPresentBooks = async () => ({
+      totalPresentBooks: await this.prisma.bookCopy.count({
+        where: {
+          book: {
+            retailLocationId,
+          },
+          returnedAt: null,
+          AND: [
+            {
+              OR: this.noProblemsFilter,
+            },
+            {
+              OR: this.notSoldOrRefunded,
+            },
+          ],
+        },
+      }),
+    });
+
+    const getTotalSoldBooks = async () => ({
+      totalSoldBooks: await this.prisma.sale.count({
+        where: {
+          refundedAt: null,
+          bookCopy: {
+            book: {
+              retailLocationId,
+            },
+          },
+        },
+      }),
+    });
+
+    const getTotalReservedBooks = async () => ({
+      totalReservedBooks: await this.prisma.reservation.count({
+        where: {
+          book: {
+            retailLocationId,
+          },
+          deletedAt: null,
+        },
+      }),
+    });
+
+    const getTotalRequestedBooks = async () => ({
+      totalRequestedBooks: await this.prisma.bookRequest.count({
+        where: {
+          book: {
+            retailLocationId,
+          },
+          deletedAt: null,
+        },
+      }),
+    });
+
+    const getTotalUsers = async () => ({
+      totalUsers: await this.prisma.user.count({
+        where: {
+          OR: [
+            {
+              requestedBooks: {
+                some: {
+                  book: {
+                    retailLocationId,
+                  },
+                  deletedAt: null,
+                },
+              },
+            },
+            {
+              bookCopies: {
+                some: {
+                  book: {
+                    retailLocationId,
+                  },
+                },
+              },
+            },
+          ],
+        },
+      }),
+    });
+
+    const getActiveSales = async () => ({
+      activeSales: await this.prisma.sale.findMany({
+        where: {
+          refundedAt: null,
+          bookCopy: {
+            book: {
+              retailLocationId,
+            },
+          },
+        },
+        include: {
+          bookCopy: {
+            select: {
+              settledAt: true,
+              book: {
+                select: {
+                  originalPrice: true,
+                },
+              },
+            },
+          },
+        },
+      }),
+    });
+
+    const getTotals = async () => {
```

Comment:
This may be the only case for which this pattern, using an arrow function to generate a promise, could make sense

Reply:
Agreed, left it there